### PR TITLE
[JENKINS-42561] Add a param to the timeout on some hang-prone operations in the CPS Vm thread

### PR DESCRIPTION
**Pull request by lucasam**
[JENKINS-42561](http://issues.jenkins-ci.org/browse/JENKINS-42561)

### CODE REVIEW

**Comment by jglick** [comment]
How does this relate to #114?

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java [hunk 1]
```diff
@@ -314,4 +314,13 @@ public static CpsThread current() {
...
         return "Thread #" + id + String.format(" @%h", this);
     }
+
+    /**
+     * Gets a the timeout collect from a System property or default value.
+     * Use -Dorg.jenkinsci.plugins.workflow.cps.CpsThread.runTimeoutMinutes
+     * @return
+     */
+    public long getRunTimeoutMinutes() {
+        return Long.parseLong(System.getProperty(getClass().getName() + ".runTimeoutMinutes", "5"));
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java]: `Long.getLong(String, long)` is what you meant.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java [hunk 2]
```diff
@@ -314,4 +314,13 @@ public static CpsThread current() {
         // getExecution().getOwner() would be useful but seems problematic.
         return "Thread #" + id + String.format(" @%h", this);
     }
+
+    /**
+     * Gets a the timeout collect from a System property or default value.
+     * Use -Dorg.jenkinsci.plugins.workflow.cps.CpsThread.runTimeoutMinutes
+     * @return
+     */
+    public long getRunTimeoutMinutes() {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java]: Typically such tuning constants are `public static /* not final */` fields, for ease of manipulation from `/script` etc.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java (hunk 1 above)
Comment (lucasam) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java]: it wasn't. But now it is :+1:

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java (hunk 2 above)
Comment (lucasam) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java]: changed to a class variable

**Pull request closed without merging**